DEVELOPING.md

```diff
@@ -121,7 +121,6 @@ make docs
 
 Upgrading Go version for the project requires the following changes:
 - Change Go version in [go.mod file](https://github.com/UpCloudLtd/terraform-provider-upcloud/blob/v2.4.1/go.mod)
-- Change `go-version` argument for `Setup Go` step in [acceptance test GitHub action file](https://github.com/UpCloudLtd/terraform-provider-upcloud/blob/v2.4.1/.github/workflows/terraform.yml)
```

Comment:
Don't need this anymore, go version is now sourced from go.mod file for those actions